pkg/apis/tenancy/v1alpha1/helper/helper_test.go

```diff
@@ -86,14 +86,20 @@ func TestParseLogicalClusterName(t *testing.T) {
 		expectedErr  bool
 	}{
 		{
-			name:         "valid name",
+			name:         "valid name for organization workspace",
```

Comment:
I don't think this is right - an organization would be `admin_organization`, not `organization`. The only logical cluster without `_` is `admin` or whatever we choose the root to be.

Reply:
Ok, so we could have

- /clusters/admin - org=?, ws=?
- /clusters/foobar - invalid (?)
- /clusters/admin_my-org - org=admin, ws=my-org
- /clusters/my-org_my-ws - org=my-org, ws=my-ws

It sounds like you're saying /clusters/foobar is invalid. If we all agree, I can update to check for that.

For /clusters/admin, what should this parsing helper return for org & ws?

Reply:
I assume admin = root in the future. So root_org makes sense technically, but is guly. And yes, foobar is invalid.

Admin aka root will be a special case forever. Only hack we could do is root_root.

Reply:
And we have to think about the local system lclusters. They might be different, i.e. some more exception. Or we find a naming scheme for them that fits into what we have here: e.g. local_foo.

Reply:
How does this look?
```go
{
			name:         "request for /clusters/admin",
			input:        "admin",
			expectedOrg:  "admin",
			expectedName: "admin",
		},
		{
			name:         "valid name for organization workspace",
			input:        "admin_organization",
			expectedOrg:  "admin",
			expectedName: "organization",
		},
		{
			name:         "valid name for org and workspace",
			input:        "organization_workspace",
			expectedOrg:  "organization",
			expectedName: "workspace",
		},
		{
			name:        "invalid name",
			input:       "foobar",
			expectedErr: true,
		},
```